portal-api/server/v1/gef/routes.js

```diff
@@ -41,76 +41,574 @@ const {
 
 // Application
+/**
+ * @openapi
+ * /application:
+ *  get:
+ *    summary: Get all applications
+ *    tags: [Applications]
+ *    responses:
+ *      200:
+ *        description: List of applications
+ * post:
+ *    summary: Create a new application
+ *    tags: [Applications]
+ *    parameters:
+ *      - in: body
+ *        name: body
+ *        description: The application to create.
+ *        schema:
+ *          type: object
+ *    responses:
+ *      201:
+ *        description: Application created successfully
+ *      400:
+ *        description: Invalid Id
+ *      422:
+ *        description: Validation Errors
+ *      500:
+ *       description: Internal Server Error
+ */
 router
   .route('/application')
   .get(validateUserHasAtLeastOneAllowedRole({ allowedRoles: [MAKER, CHECKER, READ_ONLY, ADMIN] }), application.getAll)
   .post(validateUserHasAtLeastOneAllowedRole({ allowedRoles: [MAKER] }), application.create);
 
+/**
+ * @openapi
+ * /application/clone:
+ *  post:
+ *    summary: Clone an existing application
+ *    tags: [Applications]
+ *    parameters:
+ *      - in: body
+ *        name: body
+ *        description: The application to clone.
+ *        schema:
+ *          type: object
+ *    responses:
+ *      201:
+ *        description: Application cloned successfully
+ *      400:
+ *       description: Invalid Id
+ *      404:
+ *       description: Application to clone not found
+ *      500:
+ *       description: Internal Server Error
+ */
 router.route('/application/clone').post(validateUserHasAtLeastOneAllowedRole({ allowedRoles: [MAKER] }), cloneApplication.clone);
 
+/**
+ * @openapi
+ * /application/:id:
+ *  get:
+ *    summary: Get application by ID
+ *    tags: [Applications]
+ *    parameters:
+ *      - in: path
+ *        name: id
+ *        required: true
+ *        description: The ID of the application to retrieve
+ *        schema:
+ *          type: string
+ *    responses:
+ *      200:
+ *        description: Application retrieved successfully
+ *      400:
+ *        description: Invalid Id
+ *      204:
+ *        description: No Content
+ *  put:
+ *    summary: Update application by ID
+ *    tags: [Applications]
+ *    parameters:
+ *      - in: path
+ *        name: id
+ *        required: true
+ *        description: The ID of the application to update
+ *        schema:
+ *          type: string
+ *      - in: body
+ *        name: body
+ *        description: The application data to update.
+ *        schema:
+ *          type: object
+ *    responses:
+ *      200:
+ *        description: Application updated successfully
+ *      400:
+ *        description: Invalid Id
+ *      404:
+ *        description: Application to update not found
+ *      422:
+ *        description: Validation Errors
+ *      500:
+ *        description: Internal Server Error
+ * delete:
+ *    summary: Delete application by ID
+ *    tags: [Applications]
+ *    parameters:
```

Comment:
@zainUKEF I think this is req.params not the req.body. You have to add something like this ? At least this how I dit it. 
```suggestion
 *       requestBody:
 *       description: Fields required to create a deal. Creates other default fields
 *       required: true
 *       content:
 *         application/json:
 *           schema:
 *             type: object
 *             properties:
 *               user:
 *                 type: object
 *               deal:
 *                 type: object
 *                 properties:
 *                   details:
 *                     type: object
 *               auditDetails:
 *                 type: object
 *                 $ref: '#/definitions/PortalAuditDetails'
 *           example:
 *             bankInternalRefName: 'a1'
 *             additionalRefName: 'test'
 *             bank: { id: '9' }
 *             maker: { _id: '123abc' }
 *             details:
 *             ukefDealId: '20010739'
 *             submissionCount: 1
```